x-pack/plugins/enterprise_search/server/routes/app_search/app_search_request_handler.test.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mockConfig, mockLogger } from '../__mocks__';
+
+import { createAppSearchRequestHandler } from './app_search_request_handler';
+
+jest.mock('node-fetch');
+const fetch = jest.requireActual('node-fetch');
+const { Response } = fetch;
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fetchMock = require('node-fetch') as jest.Mocked<typeof fetch>;
+
+const responseMock = {
+  ok: jest.fn(),
+  notFound: jest.fn(),
+};
+
+beforeEach(() => {
+  responseMock.ok.mockClear();
+  responseMock.notFound.mockClear();
+  fetchMock.mockReset();
+});
+
+describe('createAppSearchRequestHandler', () => {
```

Comment:
Quick request to nest this in the main describe(), under line 33

```suggestion
describe('createAppSearchRequestHandler', () => {
  const responseMock = {
    ok: jest.fn(),
    notFound: jest.fn(),
  };

  beforeEach(() => {
    responseMock.ok.mockClear();
    responseMock.notFound.mockClear();
    fetchMock.mockReset();
  });
```